### ISSUE TITLE
MacOS Deployment target issues when building resvg as a static library

### ISSUE THREAD

**Issue by JanosGit** [RazrFalcon/resvg#328]
First of all, thank you for this great library! I use resvg in a MacOS audio plugin (C++ project) via the C-API. As static linked libraries make this plugin quite a bit more portable, I modified the Cargo.toml file to output a static library, everything else has been left unmodified.

My project has a MacOS deployment target of 10.9. I know that the default Rust deployment target is 10.7, so that should be fine. But linking to resvg leads to a lot of linker warning like 
```
ld: warning: object file (libresvg.a(skia_c.o)) was built for newer macOS version (10.15) than being linked (10.9)
ld: warning: object file (libresvg.a(SkMemory_malloc.o)) was built for newer macOS version (10.15) than being linked (10.9)
ld: warning: object file (libresvg.a(SkDebug_stdio.o)) was built for newer macOS version (10.15) than being linked (10.9)
```

If I get it right, this is all about Skia symbols. As the Skia part of your library is not rust based, is it possible that no explicit MacOS deployment target is specified for the Skia build (which usually leads to the build machines architecture being taken as the deployment target on MacOS)?

**Comment by RazrFalcon** [comment]
Try using the master branch. It no longer uses Skia.

**Comment by JanosGit** [comment]
Thank you for the quick reply. I pulled the latest master and rebuilt the library, now the warnings for Skia symbols disappeared, however I find a lot of other similar warnings, e.g. 
```
ld: warning: object file (libresvg.a(hb-blob.o)) was built for newer macOS version (10.15) than being linked (10.9)
ld: warning: object file (libresvg.a(hb-buffer.o)) was built for newer macOS version (10.15) than being linked (10.9)
```

**Comment by RazrFalcon** [comment]
I'm not familiar with mac, so I'm not sure how to fix this issue. `hb-*` is from harfbuzz, which we're linking statically. And since we're using [cc](https://crates.io/crates/cc) I think you should ask them for directions.

By the way, resvg works fine on Catalina. I'm building `tools/svgview` just fine.

**Comment by JanosGit** [comment]
> By the way, resvg works fine on Catalina. I'm building tools/svgview just fine.

Yes I can confirm that :) The problem arises when using a Catalina build machine (like I do) and then execute the built binary on an older Mac, so I got reports from crashes on High Sierra users. Therefore on MacOS you usually explicitly specify the deployment target for building, meaning the minimum MacOS version the resulting binary should be able to run on. If you don't do so, it will just assume that you are building for a minimum deployment target equaling the build machine and I have the feeling that this happens here.

Being a C++ developer and working with CMake most of the time, I'm having a hard time right now figuring out how the rust tools manage the build, could you give me a more detailed pointer about where the build of harfbuzz is triggered? Maybe I can figure out some additional flags to get this right.

**Comment by JanosGit** [comment]
I guess this is the place where the harfbuzz build happens: https://github.com/RazrFalcon/rustybuzz/blob/master/build.rs? 

Probably adding a `-mmacosx-version-min=10.7` flag in case you are building for MacOS should already be enough to get things working. I'd recommend chasing 10.7 as it is the default deployment target for rust if I got it all right

**Comment by RazrFalcon** [comment]
Can you test it locally? Just download the rustybuzz, update build.rs and then replace

```
rustybuzz = { version = "0.1", optional = true }
```

with 

```
rustybuzz = { path = "/path/to/rustybuzz", optional = true }
```

in `resvg/usvg/Cargo.toml`

**Comment by JanosGit** [comment]
I tried that out. Without doing any modification to rustybuzz I first had to check out v 0.1.0 in order to make it compile with the local rustybuzz clone. Unfortunately, adding `build.flag("-mmacosx-version-min=10.7");` did not affect the build as I assumed, in some stackoverflow posts I read about that this flag should be passed to the linker as well in order to work and I did not find out how to specify linker flags and not only compiler flags in the build.rs file of rustybuzz.

However, as a second approach I tried explicitly setting the `MACOSX_DEPLOYMENT_TARGET` environment variable to 10.7 in my build script before invoking `cargo build` which seemed to work – the linker warning disappear when linking against the library built that way.

While I'm glad that I got this solved for my use-case, I wonder if you see any way to do this internally in the build process instead of setting an environment variable externally prior to the build?
